fluidsf/calculate_advection_3d.py

```diff
@@ -45,14 +45,14 @@ def calculate_advection_3d(  # noqa: D417
     dz = np.abs(z[0] - z[1])
 
     if scalar is not None:
-        dsdz, dsdy, dsdx = np.gradient(scalar, dz, dy, dx, axis=(0, 1, 2))
+        dsdz, dsdy, dsdx = np.gradient(scalar, dx, dy, dz, axis=(0, 1, 2))
```

Comment:
@cassidymwagner I'm not sure this re-ordering is correct.

For example, I tested the function in a simple notebook and artificially multiplying the first element (dx) by a factor of 10 decreases the first gradient term (dsdz) by a factor of 10. I think the original order was correct as changing dx should affect dsdx (the third term), while dsdz should only be affected by changes in dz.

Are you able to test whether the new (or old) ordering produces the correct result?

Reply:
I see what you mean here, and I'm getting the same results when I adjust dx/dy/dz. I think it just needs to be consistent. Either we revert this commit back to what it was before or we could change it to be a little more intuitive: 
`dsdx, dsdy, dsdz = np.gradient(scalar, dx, dy, dz, axis=(2, 1, 0))`

Since we've been using dx=dy values we haven't run into this exact problem whereas previously the `axis` argument was causing issues. `axis=(2, 1, 0)` should mean axes x, y, and then z.

Reply:
Ok, I just made a commit following your suggestion, and one that made the 2D advection calculation ordering consistent with this 3D one